File: BackendAPI/src/main/java/com/aguare/backendapi/Entitys/ListProducts.java
package com.aguare.backendapi.Entitys;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "list_products")
public class ListProducts implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idListProducts;

    @Column(name = "fk_id_sale")
    private Long idSale;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "fk_id_product")
    private Product idProduct;

    @Column(name = "fk_id_branch")
    private Long idBranch;

    private int quantity;
    private Double subtotal;
}

File: FrontApp/src/app/Entitys/ListProducts.ts
import { Product } from "./Product";

export class ListProducts {
  idListProducts: number;
  idSale: number;
  idProduct: Product;
  quantity: number;
  subTotal: number;
  idBranch: number;

  constructor(
    idLIstProducts: number,
    idSale: number,
    idProduct: Product,
    quantity: number,
    subTotal: number,
    idBranch: number
  ) {
    this.idListProducts = idLIstProducts;
    this.idSale = idSale;
    this.idProduct = idProduct;
    this.quantity = quantity;
    this.subTotal = subTotal;
    this.idBranch = idBranch;
  }
}
